app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -165,6 +166,34 @@
     }
   }
 
+  function autocompleteFormatting(view, transaction) {
+    if (transaction.transactions.every(tr => ["input.complete"].includes(tr.annotation(Transaction.userEvent)))) {
+      indentMultilineInserts(view, transaction)
+      insertSemicolon(view, transaction)
+    }
+  }
+
+  function insertSemicolon(view, transaction) {
+    const line = view.state.doc.lineAt(transaction.changedRanges[0].fromB)
+    const phrase = getPhraseFromPosition(line, transaction.changedRanges[0].fromB)
+
+    const possibleValues = get(completionsMap).filter(v => v.args_length)
+    const validValue = possibleValues.find(v => v.label == phrase.text)
+
+    if (validValue?.type) {
+      const insertPosition = view.state.selection.ranges[0].from
+      
+      if(validValue.type === "function") {
+        view.dispatch({
+          changes: { from: insertPosition, insert: ";" },
+          selection: EditorSelection.create([
+            EditorSelection.range(insertPosition + 1, insertPosition + 1)
+          ])
+        })
+      }
+    }
```

Comment:
Perhaps we can make both of these early returns to make it a bit less pyramid-y

```suggestion
    if (!validValue?.type) return

    const insertPosition = view.state.selection.ranges[0].from
      
    if(validValue.type !== "function") return
    
    view.dispatch({
      changes: { from: insertPosition, insert: ";" },
      selection: EditorSelection.create([
        EditorSelection.range(insertPosition + 1, insertPosition + 1)
      ])
    })
```